armi/materials/material.py

```diff
@@ -240,6 +240,37 @@ def setDefaultMassFracs(self):
         pass
 
     def setMassFrac(self, nucName: str, massFrac: float) -> None:
+        """
+        Assigns the mass fraction of a nuclide within the material.
+
+        Notes
+        -----
+        This will try to convert the provided ``massFrac`` into a float
+        for assignment. If the conversion cannot occur then an error
+        will be thrown.
+        """
+        try:
+            massFrac = float(massFrac)
+        except Exception as ee:
+            raise TypeError(
+                f"Error is converting the mass fraction of {massFrac} "
+                f"for nuclide {nucName} in {self} to a float. "
+                f"Exception: {ee}"
+            )
+        if massFrac < 0.0:
```

Comment:
I take it we've seen bad mass fractions in the wild lately?

Reply:
I haven't seen it, but I am trying to guard against something to make sure we don't have an error like that. Also, it was trying to address https://github.com/terrapower/armi/issues/1198 and I was digging into the mass fractions and how they were being set for materials.